homeassistant/components/light/xiaomi_philipslight.py

```diff
@@ -0,0 +1,215 @@
+"""
+Support for Xiaomi Philips Lights (LED Ball & Ceil).
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/light.xiaomi_philipslight/
+"""
+import asyncio
+from functools import partial
+import logging
+import math
+
+import voluptuous as vol
+
+from homeassistant.util.color import (
+    color_temperature_kelvin_to_mired as kelvin_to_mired, )
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.light import (
+    PLATFORM_SCHEMA, ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS,
+    ATTR_COLOR_TEMP, SUPPORT_COLOR_TEMP, Light, )
+
+from homeassistant.const import (CONF_NAME, CONF_HOST, CONF_TOKEN, )
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Xiaomi Philips Light'
+PLATFORM = 'xiaomi_philipslight'
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): vol.All(cv.string, vol.Length(min=32, max=32)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+REQUIREMENTS = ['python-mirobo==0.1.3']
+
+# The light does not accept cct values < 1
+CCT_MIN = 1
+CCT_MAX = 100
+
+SUCCESS = ['ok']
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the light from config."""
+    from mirobo import Ceil
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+
+    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
+    light = Ceil(host, token)
```

Comment:
Would it be possible to verify that the connection has been made here?

If a connection cannot be made, raise `PlatformNotReady` and setup will be retried automatically.

```
from homeassistant.exceptions import PlatformNotReady

raise PlatformNotReady
```